wandb/sdk/internal/sender.py

```diff
@@ -890,16 +950,117 @@ def send_stats(self, record: "Record") -> None:
         self._fs.push(filenames.EVENTS_FNAME, json.dumps(row))
         # TODO(jhr): check fs.push results?
 
+    def _output_raw_finish(self) -> None:
+        for stream, output_raw in self._output_raw_streams.items():
+            output_raw._stopped.set()
+
+            # shut down threads
+            output_raw._writer_thr.join(timeout=5)
+            if output_raw._writer_thr.is_alive():
+                logger.info("processing output...")
+                output_raw._writer_thr.join()
+            output_raw._reader_thr.join()
+
+            # flush output buffers and files
+            self._output_raw_flush(stream)
+        self._output_raw_streams = {}
+        if self._output_raw_file:
+            self._output_raw_file.close()
+            self._output_raw_file = None
+
+    def _output_writer_thread(self, stream: "StreamLiterals") -> None:
+        while True:
+            output_raw = self._output_raw_streams[stream]
+            if output_raw._queue.empty():
+                if output_raw._stopped.is_set():
+                    return
+                time.sleep(0.5)
+                continue
+            data = []
+            while not output_raw._queue.empty():
+                data.append(output_raw._queue.get())
+            if output_raw._stopped.is_set() and sum(map(len, data)) > 100000:
+                logger.warning("Terminal output too large. Logging without processing.")
+                self._output_raw_flush(stream)
+                for line in data:
+                    self._output_raw_flush(stream, line)
+                # TODO: lets mark that this happened in telemetry
```

Comment:
address this todo?

Reply:
in a future PR when there is more time to focus on telemetry.